python/ray/tune/search/vizier/vizier_search.py

```diff
@@ -0,0 +1,248 @@
+        metric: The training result objective value attribute. If None
+            but a mode was passed, the anonymous metric `_metric` will be used
+            per default.
+        mode: One of {min, max}. Determines whether objective is
+            minimizing or maximizing the metric attribute.
+        algorithm: Specific algorithm from Vizier's library to use. "DEFAULT" corresponds to GP-UCB-PE. See 
+            https://oss-vizier.readthedocs.io/en/latest/guides/user/supported_algorithms.html for more options.
+
+    Tune automatically converts search spaces to Vizier's format:
+
+    .. code-block:: python
+
+        from ray import tune
+        from ray.tune.search.bayesopt import VizierSearch
+
+        config = {
+            "width": tune.uniform(0, 20),
+            "height": tune.uniform(-100, 100)
+        }
+
+        vizier = VizierSearch(metric="mean_loss", mode="min")
+        tuner = tune.Tuner(
+            my_func,
+            tune_config=tune.TuneConfig(
+                search_alg=vizier,
+            ),
+            param_space=config,
+        )
+        tuner.fit()
+
+    Alternatively, you can pass a Vizier `SearchSpace` object manually to the
+    Searcher:
+
+    .. code-block:: python
+
+        from ray import tune
+        from ray.tune.search.vizier import VizierSearch
+        from vizier import pyvizier as vz
+
+        search_space = vz.SearchSpace()
+        search_space.root.add_float_param('w', 0.0, 5.0)
+
+        vizier = VizierSearch(search_space, metric="mean_loss", mode="min")
+        tuner = tune.Tuner(
+            trainable_function,
+            tune_config=tune.TuneConfig(
+                search_alg=vizier
+            )
+        )
+        tuner.fit()
+
+    """
+
+    def __init__(
+        self,
+        space: Optional[Union[Dict, "svz.SearchSpace"] = None,
+        metric: Optional[str] = None,
+        mode: Optional[str] = None,
+        algorithm: Optional[str] = 'DEFAULT',
+    ):
+        assert IMPORT_SUCCESSFUL, 'Vizier must be installed with `pip install google-vizier[jax]`.'
+        super(VizierSearch, self).__init__(metric=metric, mode=mode)
+        self._algorithm = algorithm
+
+        # For Vizier to identify the unique study.
+        self._study_id = f'ray_vizier_{uuid.uuid1()}'
+
+        # Mapping from Ray trial id to Vizier Trial client.
+        self._active_trials: Dict[str, clients.Trial] = {}
+
+        # Vizier service client.
+        self._study_client: Optional[clients.Study] = None
+
+        if isinstance(space, dict) and space:
+            resolved_vars, domain_vars, grid_vars = parse_spec_vars(space)
+            if resolved_vars:
+                raise TypeError(SPACE_ERROR_MESSAGE)
+            if domain_vars or grid_vars:
+                logger.warning(
+                    UNRESOLVED_SEARCH_SPACE.format(par="space", cls=type(self))
+                )
```

Comment:
Just want to double check here that this warning logic is applicable to Vizier, specifically the logic in `SearchSpaceConverter`.

Reply:
I actually was never sure what these variables meant (`resolved_vars`, `domain_vars`, `grid_vars`, etc.)

I just copied this logic other algorithm codes, assuming we all shared similar supported search spaces.

Vizier only supports cartesian products of: 
1. DOUBLE: Bounded Float
2. INTEGER: Bounded Integer
3. DISCRETE: Set of floats
4. CATEGORICAL: Set of strings

Given this, do you think the warning logic should apply here?

Reply:
Actually this should be fine. I double checked the logic and this is just a warning indicating that you should be passing in the Tune search space directly into the `Tuner` rather than the `Searcher`.